pymc3/tests/test_distributions_random.py

```diff
@@ -1306,6 +1304,71 @@ class TestOrderedProbit(BaseTestDistribution):
     ]
 
 
+class TestWishart(BaseTestDistribution):
+    def wishart_rng_fn(self, size, nu, V, rng):
+        return st.wishart.rvs(nu, V, size=size, random_state=rng)
+
+    pymc_dist = pm.Wishart
+
+    V = np.eye(3)
+    pymc_dist_params = {"nu": 4, "V": V}
+    reference_dist_params = {"nu": 4, "V": V}
+    expected_rv_op_params = {"nu": 4, "V": V}
+
+    reference_dist = lambda self: functools.partial(
+        self.wishart_rng_fn, rng=self.get_random_state()
+    )
+    tests_to_run = [
+        "check_rv_size",
+        "check_pymc_params_match_rv_op",
+        "check_pymc_draws_match_reference",
+    ]
+
+
+class TestMatrixNormal(BaseTestDistribution):
+    def matrixnormal_rng_fn(self, size, mu, rowcov, colcov, rng):
+        cholesky = Cholesky(lower=True, on_error="nan")
+        rowchol = cholesky(rowcov).eval()
+        colchol = cholesky(colcov).eval()
+        size = to_tuple(size)
+        dist_shape = to_tuple([rowchol.shape[0], colchol.shape[0]])
+        output_shape = size + dist_shape
+
+        # Broadcasting all parameters
+        (mu,) = broadcast_dist_samples_to(to_shape=output_shape, samples=[mu], size=size)
+        rowchol = np.broadcast_to(rowchol, shape=size + rowchol.shape[-2:])
+
+        colchol = np.broadcast_to(colchol, shape=size + colchol.shape[-2:])
+        colchol = np.swapaxes(colchol, -1, -2)  # Take transpose
+
+        standard_normal = rng.standard_normal(output_shape)
+        samples = mu + np.matmul(rowchol, np.matmul(standard_normal, colchol))
+
+        return samples
```

Comment:
Kind-of counter-intuitive to use this function here as reference, since it's basically same as the internal one, but it looks like `st.matrix_normal.rvs(mean=mu, rowcov=rowcov, colcov=colcov, size=size)` used previously resulted in draws that are differently valued than our implementation and our previous testing just checked if the values came from similar distributions. (See previous commit https://github.com/pymc-devs/pymc3/pull/4777/commits/4ee1360158885556c7801b97bc166ae2a40c37bd).

Reply:
Does this help? https://github.com/pymc-devs/pymc3/blob/fe115b857ae42e84d1f359153fd1f7efb2c327ae/docs/source/distribution_implementation.md#notes-on-check_pymcs_draws_match_reference-test

Reply:
In this case, it may be better to use something along the lines of the `pymc3_random` test and compare with the Scipy version. Maybe compare dimension by dimension since it is a multivariate distribution.

Reply:
In that case we can add a seperate class which does this. (Similar to how we did with `Interpolated` distribution )

Reply:
You can also add it in that class to keep things organized, such as this: https://github.com/pymc-devs/pymc3/blob/f4343ca7ba112f843cf256378e88ff80a25342a6/pymc3/tests/test_distributions_random.py#L385-L390